src/app/static/src/app/utils.ts

```diff
@@ -368,19 +368,30 @@ export const validateEmail = (test: string): boolean => {
 
 export const versionLabel = (version: Version) => `v${version.versionNumber}`;
 
-export const writeOptionsIntoForm = (options: DynamicFormData, optionsForm: DynamicFormMeta) => {
-    const hasOptions = Object.keys(options).length > 0;
-    if (hasOptions) {
-        optionsForm.controlSections.forEach(section => {
-            section.controlGroups.forEach(group => {
-                group.controls.forEach(control => {
-                    if (control.name in options) {
-                        control.value = options[control.name];
-                    }
-                });
+// Parse the form payload by ensuring that types are adhered to here
+// If select or multiselect => ensure options exist
+// If multiselect => make sure value is valid
+// Then write in current options from state
+export const parseAndFillForm = (options: DynamicFormData, optionsForm: DynamicFormMeta) => {
+    const stateHasOptions = Object.keys(options).length > 0;
+    optionsForm.controlSections.forEach(section => {
+        section.controlGroups.forEach(group => {
+            group.controls.forEach(control => {
+                // Write existing option into controls
+                if (stateHasOptions && control.name in options) {
+                    control.value = options[control.name];
+                }
+                // Ensure options exist
+                if (hasOptions(control) && !control.options) {
```

Comment:
This line suggests that `hasOptions` is poorly named! Should be `isSelect`?

Reply:
it could be select or multiselect for `hasOptions` so perhaps `isDropdown` might be a more suitable name? `isSelect` sounds like the function is checking `control.type === "select"` only